src/components/Pages/RequestPage/RequestPage.tsx

```diff
@@ -65,6 +65,19 @@ export const RequestPage = () => {
         // Try to restablish connection with the wallet.
         const connectionData = await connection.tryPreviousConnection()
```

Comment:
This might not be the best way to check if the user is connected given that internally it calls the connection logic.

Given that this check is used in multiple pages, I don't know how it would behave with the different wallet providers that we support.

It has not given us any issues yet but it would be nice to check it out or at least abstract the tryPreviousConnection to when the application is initialized.